contracts/near/eth-prover/src/lib.rs

```diff
@@ -211,19 +281,23 @@ impl EthProver {
             if head % 2 == 1 {
                 path.push(path_u8[0] % 16);
             }
-            for val in path_u8.into_iter().skip(1) {
+            for val in path_u8.iter().skip(1) {
                 path.push(val / 16);
                 path.push(val % 16);
             }
-            assert_eq!(path.as_slice(), &key[key_index..key_index + path.len()]);
 
             if head >= 2 {
                 // Leaf node
                 assert_eq!(proof_index + 1, proof.len());
                 assert_eq!(key_index + path.len(), key.len());
-                get_vec(&node, 1)
+                if path.as_slice() == &key[key_index..key_index + path.len()] {
+                    get_vec(&node, 1)
+                } else {
+                    vec![]
+                }
             } else {
                 // Extension node
+                assert_eq!(path.as_slice(), &key[key_index..key_index + path.len()]);
```

Comment:
Consider the proof for a key not included in the Trie whose path diverges from other keys in the middle of an Extension node. In this scenario I expect vec![] should be returned to prove non-existence ?

Reply:
created issue to examine security implications and add it later, https://github.com/aurora-is-near/rainbow-bridge/issues/905